pkgs/development/interpreters/spidermonkey/common.nix

```
@@ -51,16 +54,24 @@ stdenv.mkDerivation (finalAttrs: rec {
   ] ++ lib.optionals (lib.versionOlder version "91") [
     autoconf213
     yasm
+  ] ++ lib.optionals stdenv.isDarwin [
+    xcbuild
+    libobjc
+    libiconv
```

Do these really go here rather than `buildInputs`?

---

@jtojnar yes, it is linked against `libiconv.dylib` on macOS:
```sh
√ result % otool -L lib/libmozjs-102.dylib     
lib/libmozjs-102.dylib:
	/nix/store/9ld6s0lcdjw9cp26zf4qa0j24m80rwd5-spidermonkey-102.0/lib/libmozjs-102.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.60.2)
	/nix/store/dpfzpp8c7z0fdxxvin21pk5gs135wn39-Libsystem-1238.60.2/lib/libresolv.9.dylib (compatibility version 1.0.0, current version 1.0.0)
	/nix/store/k56igcml51xccxxakgvj8lai6f7fz05q-zlib-1.2.12/lib/libz.dylib (compatibility version 1.0.0, current version 1.2.12)
	/nix/store/q4c16zw3kimqdvxcc06njrv49bq68hpf-libiconv-50/lib/libiconv.dylib (compatibility version 7.0.0, current version 7.0.0)
	/nix/store/nbwbggjf37pxmishd2sqqbswv6f0l3cb-libcxx-11.1.0/lib/libc++.1.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1454.90.0)
√ result % otool -L bin/js102     
bin/js102:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.60.2)
	/nix/store/dpfzpp8c7z0fdxxvin21pk5gs135wn39-Libsystem-1238.60.2/lib/libresolv.9.dylib (compatibility version 1.0.0, current version 1.0.0)
	/nix/store/k56igcml51xccxxakgvj8lai6f7fz05q-zlib-1.2.12/lib/libz.dylib (compatibility version 1.0.0, current version 1.2.12)
	/nix/store/qgzlayc4acky2pyp56dh6p1bm00d315g-readline-6.3p08/lib/libreadline.6.dylib (compatibility version 6.0.0, current version 6.3.0)
	/nix/store/q4c16zw3kimqdvxcc06njrv49bq68hpf-libiconv-50/lib/libiconv.dylib (compatibility version 7.0.0, current version 7.0.0)
	/nix/store/nbwbggjf37pxmishd2sqqbswv6f0l3cb-libcxx-11.1.0/lib/libc++.1.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	@rpath/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1454.90.0)
√ result % 
```

---

If it is linked against a library it goes into buildInputs.

---

I think this is the last open thread.

```suggestion
    libobjc
```
libiconv is already in buildInputs. Can we mopve libobjc, too?